### PR TITLE
Speed up CSV generation examples.

### PR DESCRIPTION
**Pull request by axlan**
Improve the speed of the CSV write step by 1000x. The slowest step is the `reader.read()` call which can still take  ~5 minutes on a large log with high rate measurements. Writing the CSV now takes less than a minute.

### CODE REVIEW

File: python/examples/extract_vehicle_speed_data.py [hunk 1]
```diff
@@ -55,87 +55,75 @@
...
             RawVehicleSpeedOutput],
-        show_progress=True)
-    if all(len(d.messages) == 0 for d in result.values()):
+        show_progress=True,
+        return_numpy=True)
+    if all(len(d.p1_time) == 0 for d in result.values()):
         logger.warning('No speed data found in log file.')
         sys.exit(2)
 
     wheel_speed_data = result[WheelSpeedOutput.MESSAGE_TYPE]
```
Comment (adamshapiro0) [inline python/examples/extract_vehicle_speed_data.py:64]: Move below comment

File: python/examples/extract_vehicle_speed_data.py [hunk 2]
```diff
@@ -55,87 +55,75 @@
...
+                                    gps_time,
+                                    wheel_speed_data.front_left_speed_mps,
+                                    wheel_speed_data.front_right_speed_mps,
+                                    wheel_speed_data.rear_left_speed_mps,
+                                    wheel_speed_data.rear_right_speed_mps,
+                                    wheel_speed_data.gear], axis=1), fmt='%.6f')
     else:
         logger.info("No corrected wheel speed data.")
 
+    raw_wheel_speed_data = result[RawWheelSpeedOutput.MESSAGE_TYPE]
```
Comment (adamshapiro0) [inline python/examples/extract_vehicle_speed_data.py:82]: Move below comment

File: python/examples/extract_vehicle_speed_data.py [hunk 3]
```diff
@@ -55,87 +55,75 @@
...
+                                    gps_time,
+                                    raw_wheel_speed_data.front_left_speed_mps,
+                                    raw_wheel_speed_data.front_right_speed_mps,
+                                    raw_wheel_speed_data.rear_left_speed_mps,
+                                    raw_wheel_speed_data.rear_right_speed_mps,
+                                    raw_wheel_speed_data.gear], axis=1), fmt='%.6f')
     else:
         logger.info("No raw wheel speed data.")
 
+    vehicle_speed_data = result[VehicleSpeedOutput.MESSAGE_TYPE]
```
Comment (adamshapiro0) [inline python/examples/extract_vehicle_speed_data.py:100]: Move below comment

File: python/examples/extract_vehicle_speed_data.py [hunk 4]
```diff
@@ -55,87 +55,75 @@
...
-                     message.vehicle_speed_mps,
-                     message.gear))
+            np.savetxt(path, np.stack([vehicle_speed_data.p1_time,
+                                       gps_time,
+                                       vehicle_speed_data.vehicle_speed_mps,
+                                       vehicle_speed_data.gear], axis=1), fmt='%.6f')
     else:
         logger.info("No corrected vehicle speed data.")
 
+    raw_vehicle_speed_data = result[RawVehicleSpeedOutput.MESSAGE_TYPE]
```
Comment (adamshapiro0) [inline python/examples/extract_vehicle_speed_data.py:115]: Move below comment

**Review by adamshapiro0 — APPROVED** [review]
Looks good

File: python/examples/extract_imu_data.py [hunk 5]
```diff
@@ -49,36 +49,39 @@
 
     # Read satellite data from the file.
     reader = DataLoader(input_path)
-    result = reader.read(message_types=[IMUOutput, RawIMUOutput], show_progress=True)
+    result = reader.read(message_types=[IMUOutput, RawIMUOutput], show_progress=True, return_numpy=True)
```
[the remark this answers is not in the document]
Comment (axlan) [reply python/examples/extract_imu_data.py:52]: At least for me `extract_satellite_info.py` and `extract_position_data.py` each take less than 30 seconds and have some other complications in their generation that would make it not totally trivial. Also, it doesn't hurt to have examples with both patterns.

**Pull request merged**